bitcoin/psbt.c

```diff
@@ -86,6 +86,22 @@ struct wally_psbt *new_psbt(const tal_t *ctx, const struct wally_tx *wtx)
 	return psbt;
 }
 
+struct wally_psbt *combine_psbt(const tal_t *ctx,
+				const struct wally_psbt *psbt0,
+				const struct wally_psbt *psbt1)
+{
+	struct wally_psbt *combined_psbt;
+	tal_wally_start();
+	if (wally_psbt_clone_alloc(psbt0, 0, &combined_psbt) != WALLY_OK)
+		abort();
+	if (wally_psbt_combine(combined_psbt, psbt1) != WALLY_OK) {
+		tal_wally_end_onto(ctx, combined_psbt, struct wally_psbt);
```

Comment:
Is this the right thing to do here?  Should we free the memory immediately instead?

Reply:
I think this is right, because from the docs of `tal_wally_end_onto` I found that
this method iterates over an object allocated by libwally. So I guess it is the safe way to delete an object that wally worked on?

Reply:
This works, but leaks the combined_psbt.  You should `return tal_free(combined_psbt);` after this (of course, this code shouldn't happen at all, since the HSM said the PSBT was good!).